systemli/userli
# ✨ Add Api for dovecot user lookup and authentication

## Testing the Lua authentication script:

1. Install `lua5.3 lua-json dovecot-auth-lua`

2. Add a config file in vagrant box `/etc/dovecot/conf.d/99-env-vars.conf`:

```
import_environment=MAIL_CRYPT=2 USERLI_API_ACCESS_TOKEN=insecure USERLI_HOST=localhost DOVECOT_LUA_DEBUG=true DOVECOT_LUA_INSECURE=true
```

3. Replace `/etc/dovecot/conf.d/auth-checkpassword.conf.ext` with the following:
```
passdb {
  driver = lua
  args = file=/vagrant/checkpasswd.lua blocking=yes
}

userdb {
  driver = lua
  args = file=/vagrant/checkpasswd.lua blocking=yes
}
```

4. See https://doc.dovecot.org/2.3/admin_manual/debugging/debugging_authentication/ for useful commands
